PokeAlarm/Manager.py

```diff
@@ -946,6 +979,56 @@ def process_weather(self, weather):
             self._rule_log.info(
                 'Weather %s rejected by all rules.', weather.name)
 
+    def process_quest(self, quest):
+        # type: (Events.QuestEvent) -> None
+        """ Process a quest event and notify alarms if it passes. """
+
+        # Set the name for this event so we can log rejects better
+        quest.name = quest.stop_name + '-' + quest.reward
+
+        # Make sure that quest changes are enabled
+        if self._quest_enabled is False:
+            self._log.debug("Quest ignored: quest "
+                            "notifications are disabled.")
+            return
+
+        # Calculate distance and direction
+        if self.__location is not None:
+            quest.distance = get_earth_dist(
+                [quest.lat, quest.lng], self.__location, self.__units)
+            quest.direction = get_cardinal_dir(
+                [quest.lat, quest.lng], self.__location)
+
+        # Store copy of cache info
+        # TODO
```

Comment:
I think you'll want something like the following: 

```
previous_reward = self._cache.quest_reward(quest.stop_id)
if quest.reward == previous-reward:
    self._log.debug("Quest ignored: Reward previously alerted!")
    return

self._cache.quest_reward(stop_id, quest.reward)
```

But it's worth noting it the Reward is a string, then it'll probably be wrong a lot. If possible we should try to represent the Reward as an enum.